### PR TITLE
Track status of volumesnapshots to update backup via backup_operations_controller

### PR DESCRIPTION
**Pull request by kaovilai**
Signed-off-by: Tiger Kaovilai <tkaovila@redhat.com>

Thank you for contributing to Velero!

# Please add a summary of your change

# Does your change fix a particular issue?

Fixes #(issue)

# Please indicate you've done the following:

- [ ] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ ] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Comment by sseago** [comment]
@kaovilai Note that if you're tracking this during the backup operations controller, you'll need to do the first pass in the backup controller. The first pass of calling the Progress method on all async operations happens at the end of the InProgress processing in the backup controller immediately following completion of item backup. If everything is done at this point (which may happen if there aren't many volumes, volumes are small-ish, and there are a lot of non-PVC items backed up after pods/pvcs), then the WaitingForPluginOperations phase is skipped entirely, and the backup moves straight to Finializing. In this case, the bacup operations controller won't reconcile the backup at all.

**Comment by kaovilai** [comment]
> WaitingForPluginOperations phase is skipped entirely

CSI plugin is now biav2, so it should be in one of the backup item operations.

If we can get it's operationID, and that OperationProgress is completed without errors then we can consider that CSI snapshot completed successfully.

**Comment by sseago** [comment]
@kaovilai "CSI plugin is now biav2, so it should be in one of the backup item operations."

Yes -- I'm just saying that whatever you do in backup_operations_controller must also be done in backup_controller after the first Progress check, since if all of the backup operations (including CSI plugin operations) have already completed when the transition from InProgress happens, then the backup goes straight to finalizing, since there's no need to poll the backup for progress later, since we're already done.

In other words, Velero calls Progress repeatedly for each incomplete operation until they're all either completed or failed, but the *first* polling happens from the backup controller before leaving the InProgress phase, right before we upload the backup tarball the first time. This way if the operations are all quick, we don't need to wait/poll, so the WaitingForPluginOperations phase is skipped completely.

**Comment by kaovilai** [comment]
Thanks for the note
